### PR TITLE
sql/mysql: support constant literals in default values

### PR DESCRIPTION
**Pull request by a8m**

### CODE REVIEW

File: sql/mysql/migrate.go [hunk 1]
```diff
@@ -424,6 +416,30 @@ func (*state) attr(b *sqlx.Builder, attrs ...schema.Attr) {
...
+		if !hasNumericDefault(c.Type.Type) && !isHex(v) {
+			v = quote(v)
+		}
+		b.P("DEFAULT", v)
+	case *schema.RawExpr:
+		v := x.X
+		// For backwards compatibility, quote raw expressions that are not wrapped
+		// with parens for non-numeric column types (i.e. literals).
+		switch t := c.Type.Type; {
+		case isHex(v), hasNumericDefault(t), strings.HasPrefix(v, "(") && strings.HasSuffix(v, ")"):
```
Comment (yonidavidson) [inline sql/mysql/migrate.go:433]: is this just for future use?

**Review by yonidavidson — APPROVED** [review]
LGTM, 1 comment

**Review by rotemtam — APPROVED** [review]

File: sql/mysql/migrate.go (hunk 1 above)
Comment (a8m) [reply sql/mysql/migrate.go:433]: No, it is to avoid adding their negation in the if below. It will make it too long.

**Pull request merged**